src/Cache/CacheItem.php

```diff
@@ -105,21 +99,15 @@ public function expiresAt($expiration): self {
   /**
    * {@inheritdoc}
    */
-  public function expiresAfter($time): self {
+  public function expiresAfter($time): static {
     if ($time === NULL) {
       $this->expiry = NULL;
     }
-    elseif ($time instanceof DateInterval) {
-      $this->expiry = microtime(TRUE) + DateTime::createFromFormat('U', 0)->add($time)->format('U.u');
-    }
-    elseif (is_int($time)) {
-      $this->expiry = $time + microtime(TRUE);
+    elseif ($time instanceof \DateInterval) {
+      $this->expiry = microtime(TRUE) + (float) \DateTime::createFromFormat('U', '0')->add($time)->format('U.u');
     }
     else {
-      throw new \RuntimeException(sprintf(
-        'Expected $time to be either an integer, an instance of DateInterval or null, got %s',
-        is_object($time) ? get_class($time) : gettype($time)
-      ));
+      $this->expiry = $time + microtime(TRUE);
```

Comment:
Why has the type check (`is_int`) been removed?

Reply:
According to the interface definition $time can only be DateInterval|int|null.
https://github.com/php-fig/cache/blob/3.0.0/src/CacheItemInterface.php